### PR TITLE
Add donate to D Language foundation page

### PR DESCRIPTION
**Pull request by wilzbach**
I gave it a try ;-)

### CODE REVIEW

File: donate.dd [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+We welcome contributions from both organizations and individuals. Your help is highly appreciated.
+
+$(DONATE_ITEM Donate through PayPal, paypal,
+
+    Click the PayPal button to make a donation to the D Language Foundation via PayPal:
+    $(BR) $(BR)
+
+    <form action="https://www.paypal.com/cgi-bin/webscr" method="post" target="_top">
+    <input type="hidden" name="cmd" value="_s-xclick" />
+    <input type="hidden" name="hosted_button_id" value="6DV3SYW3A5ZPG" />
```
Comment (andralex) [inline donate.dd]: OK. There are a few more pages:

* Return URL (after the donation is made
* Canceled checkout URL
* Return to Merchant Link
* URL for IPN messaging - no idea what that is

For now I didn't customize any of these. I got this code:

```html
<form action="https://www.paypal.com/cgi-bin/webscr" method="post" target="_top">
<input type="hidden" name="cmd" value="_s-xclick" />
<input type="hidden" name="hosted_button_id" value="K8PLNUP7276BQ" />
<input type="image" src="https://www.paypal.com/en_US/i/btn/btn_donateCC_LG.gif" border="0" name="submit" title="PayPal - The safer, easier way to pay online!" alt="Donate" />
<img alt="" border="0" src="https://www.paypal.com/en_US/i/scr/pixel.gif" width="1" height="1" />
</form>
```

and the following url: https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=K8PLNUP7276BQ

File: donate.dd [hunk 2]
```diff
@@ -0,0 +1,91 @@
+Ddoc
+
+$(D_S $(TITLE),
+
+$(H4 The $(LINK2 $(ROOT_DIR)foundation.html, D Language Foundation) is an organization devoted to advancing
+     open source technology related to the D programming language.
+)
+
+The D Language foundation:
```
Comment (andralex) [inline donate.dd]: capital F

File: donate.dd [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+$(COMMENT keep this list in sync with foundation.dd)
+$(UL
+    $(LI holds and defends the copyright and other intellectual property rights for the D programming language.)
+    $(LI provides resources for furthering the development and adoption of the D programming language.)
+    $(LI awards $(LINK2 http://dlang.org/blog/2016/12/05/the-d-language-foundations-scholarship-program, scholarships)
```
Comment (andralex) [inline donate.dd]: Use HTTP?

(sorry couldn't resist :))

File: donate.dd [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        to highly skilled students.)
+    $(LI organizes the yearly $(HTTP dconf.org, D conference).)
```
Comment (andralex) [inline donate.dd]: GAAAAAAA!

File: donate.dd [hunk 5]
```diff
@@ -0,0 +1,91 @@
+Ddoc
+
+$(D_S $(TITLE),
+
+$(H4 The $(LINK2 $(ROOT_DIR)foundation.html, D Language Foundation) is an organization devoted to advancing
+     open source technology related to the D programming language.
```
Comment (andralex) [inline donate.dd]: Add this text: The Foundation is a $(HTTPS en.wikipedia.org/wiki/501(c)_organization, 501(c) non-profit public charity) registered in the state of Washington, USA.

File: donate.dd (hunk 4 above)
Comment (andralex) [inline donate.dd]: s/D conference/DConf, the international annual conference of D language developers and users/

File: donate.dd [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+$(P If you have a specific target or project that you wish to directly support, please $(LINK2 mailto:foundation@dlang.org, contact the D Language Foundation).)
+
+$(P By the Foundation bylaws, its officers (Walter Bright, Andrei Alexandrescu and Ali Çehreli)
+    cannot get income from the Foundation, which preempts a variety of conflicts of interest.)
+
+$(H3 The D Language Foundation is a non-profit organization)
+
+$(P For US taxpayers, contributions to the D Language Foundation are tax-deductible.
+The D Language Foundation has been recognized as a public charity under section 501(c)(3) of the Internal Revenue Code.
+You should consult your tax adviser for rules as they apply to your situation.)
```
Comment (andralex) [inline donate.dd]: Ah, I see you do have this now. You may want to move the public charity part up the page. Leave here: For US taxpayers, contributions to the D Language Foundation are tax-deductible. You should consult your tax adviser for rules as they apply to your situation.

**Comment by wilzbach** [comment]
>> @andralex - could you please go at
https://www.paypal.com/buttons/paymentbuttons/donate/create and generate such a button?
> OK. There are a few more pages:
> [..]
> For now I didn't customize any of these. I got this code:

So I just tested the button and I think it would be nice if we enable the message box:

image

It could be sth. like "Your message to the D Foundation".

**Comment by andralex** [comment]
OK, here's the code:

```html
<form action="https://www.paypal.com/cgi-bin/webscr" method="post" target="_top">
<input type="hidden" name="cmd" value="_s-xclick" />
<input type="hidden" name="hosted_button_id" value="FDEUVLD5MB44G" />
<input type="image" src="https://www.paypal.com/en_US/i/btn/btn_donate_LG.gif" border="0" name="submit" title="PayPal - The safer, easier way to pay online!" alt="Donate" />
<img alt="" border="0" src="https://www.paypal.com/en_US/i/
```

and the link:

https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=FDEUVLD5MB44G

I see every test is costing you $5 :)

**Comment by andralex** [comment]
I also eliminated the credit card logos to make it more discreet

File: foundation.dd [hunk 7]
```diff
@@ -2,8 +2,10 @@ Ddoc
 
 $(D_S $(TITLE),
 
-$(H4 The D Language Foundation is an organization devoted to advancing
-     open source technology related to the D programming language.
+$(H4 The D Language Foundation is a
+    $(HTTPS en.wikipedia.org/wiki/501(c)_organization, 501(c) non-profit public charity)
+     organization devoted to advancing open source technology
```
Comment (andralex) [inline foundation.dd]: "organization" not needed anymore

**Comment by wilzbach** [comment]
> OK, here's the code:

Thanks! Btw I think it's also possible to edit existing buttons.

Here's how it looks for a donor:



image

image

Maybe we can put an address into the PayPal account, s.t. the automatically generated receipt has one?

File: foundation.dd [hunk 8]
```diff
@@ -39,29 +46,45 @@ $(UL
...
+
+    $(BR)
+    $(DIVC dlang-address,
+        $(B The D Language Foundation)
+        $(BR)
+        13868 65th Place NE
+        $(BR)
+        Kirkland, WA 98034-4900
+        $(BR)
+    )
```
Comment (andralex) [inline foundation.dd]: He asked to not make it even easier, so please remove. cc @WalterBright

Comment (wilzbach) [inline foundation.dd]: Oh it seems that I didn't get the mail - I have rebased the PR, s.t. it's not included in the history.

**Comment by andralex** [comment]
nice!

**Pull request merged**